File: Programmers/level-1/소수만들기.py
from itertools import combinations

def isprime(x):
    # 2부터 x//2까지 나눠서 하나라도 나누어 떨어지는 경우 소수가 아니라고 판단.
    for num in range(2,x//2+1):
        if x%num == 0:
            return False
    return True

def solution(nums):
    answer = 0

    # 길이가 4라면 0,1까지만
    arr = list(combinations(nums, 3))
    for n1, n2, n3 in arr:
        if isprime(n1+n2+n3):   
            answer += 1
    return answer

File: java/프로그래머스/level-1/소수만들기.java
class Solution {
    public int solution(int[] nums) {
        int answer = 0;

        for (int i =0; i< nums.length; i++){
            for (int j =i+1; j< nums.length; j++){
                for (int k =j+1; k< nums.length; k++){
                    if (sosu(nums[i]+nums[j]+nums[k])){
                        answer++;
                    }
                }   
            }   
        }
        return answer;
    }
    
    public static boolean sosu(int sum){
        if (sum ==2){
            return true;
        }
        for (int i =2; i < sum; i++){
            if (sum %i==0){
                return false;
            }
        }
        return true;
    }
}